RELEASE_NOTES.md

```diff
@@ -1,3 +1,35 @@
+## WAMR-2.3.1
+
+### Breaking Changes
+
+- Revert the location to install public headers (#4295)
```

Comment:
i guess it's more user friendly to mention that this restores the compatibility with WAMR-2.2.0 and earlier.